src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -187,6 +176,33 @@ public TaskResult call() {
         return new TaskResult(exception);
     }
 
+    private void createLeasesForChildShardsIfNotExist() throws InvalidStateException, DependencyException, ProvisionedThroughputException {
+        for (ChildShard childShard : childShards) {
+            final String leaseKey = childShard.getShardId();
+            if (leaseCoordinator.getCurrentlyHeldLease(leaseKey) == null) {
```

Comment:
Shouldn't we be using LeaseRefresher#getLease? LeaseCoordinator#getCurrentlyHeldLease only returns the cached in-memory lease; I believe this will not work if another worker holds this lease.

Reply:
If another worker holds the lease, then the current worker should not try to create child lease for this shard right?

Reply:
More specifically for a case where a worker holds the current lease, but one of the child leases is held by another worker.

Reply:
Ah yeah, I get what you mean. I will make the change